src/core/json/jsont.rs

```diff
@@ -0,0 +1,102 @@
+#![allow(unused)]
+
+use std::collections::HashMap;
+
+pub trait JsonT {
+    type Output;
+    type Input;
+
+    fn array_ok(value: &Self::Input) -> Result<&Vec<Self::Output>, &str>;
+    fn str_ok(value: &Self::Input) -> Result<&str, &str>;
+    fn i64_ok(value: &Self::Input) -> Result<i64, &str>;
+    fn u64_ok(value: &Self::Input) -> Result<u64, &str>;
+    fn f64_ok(value: &Self::Input) -> Result<f64, &str>;
+    fn bool_ok(value: &Self::Input) -> Result<bool, &str>;
+    fn null_ok(value: &Self::Input) -> Result<(), &str>;
+    fn option_ok(value: &Self::Input) -> Result<Option<&Self::Output>, &str>;
+    fn get_path<'a, T: AsRef<str>>(value: &'a Self::Input, path: &'a [T]) -> Option<&'a Self::Output>;
+    fn get_key<'a>(value: &'a Self::Input, path: &'a str) -> Option<&'a Self::Output>;
+    fn new(value: &Self::Output) -> &Self;
+    fn group_by<'a>(value: &'a Self::Input, path: &'a [String]) -> HashMap<String, Vec<&'a Self::Output>>;
+}
+
+impl JsonT for serde_json::Value {
```

Comment:
```suggestion
impl JsonT for JsonTSerde {
```

We need a new type here.

Reply:
marking this as resolved